src/feature/dirauth/process_descs.c

```diff
@@ -121,6 +128,40 @@ add_fingerprint_to_dir(const char *fp, authdir_config_t *list,
   return;
 }
 
+/** Add the ed25519 key <b>edkey</b> to the smartlist of fingerprint_entry_t's
```

Comment:
Optional nitpick: Any chance of adding some unittests for this new function, or this new functionality altogether?